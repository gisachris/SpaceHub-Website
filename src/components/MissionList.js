import React from 'react';
import PropTypes from 'prop-types';
import '../styles/profile.css';

function MissionList(props) {
  const { missionName } = props;
  return (
    <table>
      <tbody>
        <tr className="joined-mission">
          <td className="Missionsdata">{missionName}</td>
        </tr>
      </tbody>
    </table>
  );
}

MissionList.propTypes = {
  missionName: PropTypes.string.isRequired,
};
export default MissionList;
