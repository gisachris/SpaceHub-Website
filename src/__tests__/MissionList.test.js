import { render } from '@testing-library/react';
import MissionList from '../mocks/MissionList.';

describe('tests for Profile component', () => {
  test('renders correctly', () => {
    const missionData = [
      {
        id: 1,
        name: 'name 1',
        desc: 'desc 1',
        join: false,
      },
    ];

    const { container } = render(<MissionList missionData={missionData} />);
    expect(container).toMatchSnapshot();
  });
});
