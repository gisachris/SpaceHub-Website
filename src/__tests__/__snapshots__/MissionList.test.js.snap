// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`tests for Profile component renders correctly 1`] = `
<div>
  <table>
    <tbody>
      <tr
        class="joined-mission"
      >
        <td
          class="Missionsdata"
        />
      </tr>
    </tbody>
  </table>
</div>
`;
